js/tests/unit/dropdown.js

```diff
@@ -388,4 +388,23 @@ $(function () {
 
     assert.ok($dropdown.parent('.btn-group').hasClass('open'), 'dropdown menu is open')
   })
+
+  QUnit.test('should close the dropdown if the user clicks on a file-input', function (assert) {
+    assert.expect(1)
+    var dropdownHTML = '<div class="btn-group">'
+        + '<button type="button" data-toggle="dropdown">Dropdown</button>'
+        + '<ul class="dropdown-menu">'
+        + '<li><input type="file" id="fileInput"></input></li>'
```

Comment:
`</input>` is invalid, just delete it.
`<input>` is an [empty element](https://developer.mozilla.org/en-US/docs/Glossary/empty_element).